### PR TITLE
refactor(match): address previous PR comments

### PR DESCRIPTION
**Pull request by AdrianAndersen**

### CODE REVIEW

**Comment by AdrianAndersen** [comment]
[Rette kommentarer fra PRer som ble merget](https://trello.com/c/JD0AaW2l/1071-rette-kommentarer-fra-prer-som-ble-merget)

**Comment by AdrianAndersen** [comment]
[Rette kommentarer fra PRer som ble merget](https://trello.com/c/JD0AaW2l/1071-rette-kommentarer-fra-prer-som-ble-merget)

File: src/collections/match/match.schema.ts [hunk 1]
```diff
@@ -62,17 +62,26 @@ const userMatchSchema = {
...
   // if true, disallow handing the items out or in at a stand, only allow match exchange
   itemsLockedToMatch: {
     type: Boolean,
     default: undefined,
   },
   // if receiver items have overrides, the generated customer items will
   // get the deadline specified in the override instead of using the branch defined deadline
-  deadlineOverrides: [{ item: String, deadline: Date }],
+  deadlineOverrides: {
+    type: Map,
```
[the remark this answers is not in the document]
Comment (LarsSelbekk) [reply src/collections/match/match.schema.ts:82]: Actually, we can use flattenObjectIds too
```typescript
    schema.set("toJSON", {
      transform: MongooseModelCreator.transformObject,
      flattenMaps: true,
      flattenObjectIds: true,
    });
    schema.set("toObject", {
      transform: MongooseModelCreator.transformObject,
      flattenMaps: true,
      flattenObjectIds: true,
    });
```
which simplifies `transformObject`

```typescript
  public static transformObject(doc: unknown, ret?: unknown): void {
    // Mongoose isn't sure which parameter to use, so try both :/
    if (!ret && doc) ret = doc;
    if (!ret) return;
    if (Array.isArray(ret)) {
      ret.forEach((document) =>
        MongooseModelCreator.transformObject({}, document),
      );
    } else if (typeof ret === "object") {
      const document = ret as Record<string, unknown>;
      // Translate _id to id only if id does not already exist
      // (embedded documents such as BlDocument.user may have an id field which is different from the _id field)
      if ("_id" in document && !("id" in document))
        document["id"] = document["_id"];
      delete document["_id"];
      delete document["__v"];
    }
  }
```

File: src/collections/match/operations/match-generate-operation-helper.ts [hunk 2]
```diff
@@ -225,23 +227,25 @@ export function verifyMatcherSpec(
     typeof m["matchMeetingDurationInMS"] === "number" &&
     !isNaN(m["matchMeetingDurationInMS"]) &&
     isBoolean(m["includeSenderItemsFromOtherBranches"]) &&
-    Array.isArray(m["additionalReceiverItems"]) &&
-    m["additionalReceiverItems"].every(
+    typeof m["additionalReceiverItems"] === "object" &&
+    isNotNullish(m["additionalReceiverItems"]) &&
+    Object.entries(m["additionalReceiverItems"]).every(
       (entry) =>
```
Comment (LarsSelbekk) [inline src/collections/match/operations/match-generate-operation-helper.ts:233]:
```suggestion
      ([branchId, itemIds]) =>
```

File: src/collections/match/operations/match-generate-operation-helper.ts [hunk 3]
```diff
@@ -225,23 +227,25 @@ export function verifyMatcherSpec(
...
-    m["deadlineOverrides"].every(
-      (override) =>
-        typeof override["item"] === "string" &&
-        ObjectId.isValid(override["item"]) &&
-        typeof override["deadline"] === "string" &&
-        !isNaN(new Date(override["deadline"]).getTime()),
+    typeof m["deadlineOverrides"] === "object" &&
+    isNotNullish(m["deadlineOverrides"]) &&
+    Object.entries(m["deadlineOverrides"]).every(
+      (deadlineOverride) =>
```
Comment (LarsSelbekk) [inline src/collections/match/operations/match-generate-operation-helper.ts:244]:
```suggestion
      ([branchId, newDeadline]) =>
```

**Review by LarsSelbekk — CHANGES_REQUESTED** [review]

File: src/collections/match/match.schema.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (LarsSelbekk) [reply src/collections/match/match.schema.ts:82]: Actually, Mongoose doesn't *actually* seem to return `Map`s, to this is moot. The ObjectId-transformation would still be helpful, but out of scope here.

**Review by LarsSelbekk — APPROVED** [review]
I fix

**Comment by AdrianAndersen** [comment]
i approve 💯

**Pull request merged**